php/libraries/SubprojectController.class.inc

```diff
@@ -0,0 +1,252 @@
+<?php
+/**
+ * This file contains the SubprojectController class.
+ *
+ * PHP Version 7
+ *
+ * @category Controller
+ * @package  Subprojects
+ * @author   Rida Abou-Haidar <rida.abou-haidar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS;
+
+/**
+ * Base class that contains method to access the subproject of projects
+ *
+ * @category Controller
+ * @package  Subprojects
+ * @author   Rida Abou-Haidar <rida.abou-haidar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class SubprojectController
+{
+    /**
+     * Array holding existing instances
+     *
+     * @var array
+     */
+    private static $_instances = [];
+
+    /**
+     * The database connection to use
+     *
+     * @var \Database
+     */
+    protected $database;
+
+    /**
+     * The constructor
+     *
+     * @param \Database $database The database connection to use
+     */
+    public function __construct(\Database $database)
+    {
+        if (!$database->isConnected()) {
+            throw new \RuntimeException(
+                "SubprojectController - Database not connected"
+            );
+        }
+        $this->database = $database;
+    }
+
+    /**
+     * Retruns a list of configured Subprojects
+     *
+     * @return array The list of subprojects
+     */
+    public function getAllSubprojects(): array
+    {
+        $subprojectRows = $this->database->pselect(
+            "SELECT
+                SubprojectID,
+                title,
+                useEDC,
+                WindowDifference,
+                RecruitmentTarget
+            FROM subproject
```

Comment:
That select statement should be put in a class variable so it sould be used in `getAllSubprojects`, `getSubprojectByTitle` and `getSubprojectByID`.

Reply:
I don't like that, but I could maybe make a class array (for selected fields) and a tablename (for FROM clause) variable and concatenate in the different queries... if that's okai with you I'll do it

Reply:
I don't see why you need those values in distinct variables. 
What would be the difference between :
`$db->select('SELECT' . implode(',', $selectedfields) . " FROM " . $tablename);`
and 
`$db->select($sqlselect);`

Reply:
aesthetic mostly!!